pkgs/by-name/fr/freej2me/package.nix

```
@@ -0,0 +1,87 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  ant,
+  jdk,
+  stripJavaArchivesHook,
+  writeShellScriptBin,
+}:
+
+stdenv.mkDerivation {
+  pname = "freej2me";
+  version = "0-unstable-2024-02-16";
+
+  src = fetchFromGitHub {
+    owner = "hex007";
+    repo = "freej2me";
+    rev = "8b9bc8a19baf26e3d92f88934a64a32f1cbc2795";
+    hash = "sha256-FaB2qnbtJnSZbrGRYDiz3Q/uNgD4GX//VaID6jHtyeU=";
+  };
+
+  nativeBuildInputs = [
+    ant
+    jdk
+    stripJavaArchivesHook
+  ];
+
+  prePatch = ''
+    substituteInPlace src/org/recompile/freej2me/Anbu.java \
+      --replace-fail "/usr/local" "${placeholder "out"}"
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ant
+    make -C src/libretro
+
+    runHook postBuild
+  '';
+
+  preInstall = ''
+    substituteInPlace src/sdl2/Makefile \
+      --replace-fail "/usr/local" "${placeholder "out"}"
+  '';
+
+  installPhase = let
+    mk-script-bin = frondend: writeShellScriptBin frondend ''
+      if [ ''$# -ne 3 ]; then
+        echo "usage: ${frondend} [midlet] [width] [height]"
+        exit
+      fi
+
+      ${lib.getExe' jdk "java"} -jar       \
+        "@out@/share/java/${frondend}.jar" \
+        "file://$(realpath $1)" $2 $3
+    '';
+
+    mk-script-install = frondend: ''
+      install -Dm755 ${mk-script-bin frondend}/bin/${frondend} -t $out/bin
+
+      substituteInPlace $out/bin/${frondend} \
+        --replace-fail "@out@" "${placeholder "out"}"
+    '';
+
+    front-ends = [ "freej2me" "freej2me-lr" ];
+    install-scripts = builtins.map mk-script-install front-ends;
+  in ''
+    runHook preInstall
+
+    install -Dm644 build/freej2me{,-lr}.jar -t $out/share/java
```

```suggestion
    install -D --mode=744 --verbose build/freej2me{,-lr}.jar -t $out/share/java
```

644 doesn't make this executable which seems unintentional

---

I don't think java jar nees to be executable to be loaded

---

oh right, my bad!